src/main/java/com/amazon/opendistroforelasticsearch/knn/common/KNNConstants.java

```diff
@@ -26,6 +26,6 @@
     public static final String LINF = "linf";
     public static final String COSINESIMIL = "cosinesimil";
     public static final String HAMMING_BIT = "hammingbit";
-    public static final String NEGDOTPROD = "negdotprod";
+    public static final String INNER_PROD = "inner_product";
```

Comment:
Previously, we didn't put "\_" (i.e. "hammingbit"). However, I like the "\_" and I think it should be kept. Just wanted to add comment here if anyone has strong opinions against this.

Reply:
Good catch. If this is not exposed to customer, then I like the idea of adding "_"

Reply:
Sorry, I am a little confused. Which one do we want to expose to customer? 

```
"space_type": "inner_product"
```
or
```
"space_type": "innerproduct"
```

Reply:
If these values are exposed to customer, my suggestion is to keep it consistent with original ones with out underscore i.e innerproduct.  But if these values are just used internally, then we can use underscore and probably rename others as well(may be part of other Cr)